### ISSUE TITLE
Heading and converting back.

### ISSUE THREAD

**Issue by keshav234156** [publiclab/PublicLab.Editor#400]
The maximum tag of heading that can be added is <h4> .we Should increase it to <h6> and also we can't convert the text back to non-heading once it's made it to heading.We should fix it as well.

**Comment by keshav234156** [comment]
@VladimirMikulic @jywarren @emilyashley what are your views on this issue!!!

**Comment by VladimirMikulic** [comment]
I don't think that it's an issue, the user can simply manually add `#` to get h5/h6.

**Comment by keshav234156** [comment]
@VladimirMikulic Agrees with you.But that only possible if you are in Markdown mode and also know how to use it.

**Comment by VladimirMikulic** [comment]
Feel free to write a wrapper around it :)

**Comment by NitinBhasneria** [comment]
@nstjean @cesswairimu @govindgoel I think we should look up this @keshav234156 makes some point.

**Comment by cesswairimu** [comment]
sure go ahead

**Comment by jywarren** [comment]
Hi all! Sorry for slow reply, but i think here, `we can't convert the text back to non-heading once it's made it to heading` is the critical piece. It's very hard to get rid of headers! There are a variety of header creation/changing bugs and some broken Markdown often gets generated as a result. So I'd like to refocus this issue on thoroughly documenting and solving the header creation/deletion bugs using our #509 #507 workflows?

**Comment by jywarren** [comment]
Here is a GIF of pretty normal-looking header interaction; creating headers by using the header button, but also by selecting text and making it a heading with the hotkeys (cmd-D):

![editor](https://user-images.githubusercontent.com/24359/81702463-5185ee80-9439-11ea-819b-eef9b1b5c25f.gif)

**Comment by jywarren** [comment]
Here is selecting a heading and using the heading button to cycle through heading sizes. These gifs could be part of documentation, and should be covered in a test:

![editor2](https://user-images.githubusercontent.com/24359/81702571-7712f800-9439-11ea-937a-4fdcff5acd52.gif)

**Comment by jywarren** [comment]
Hmm, i'm not able to get the odd header bugs I've seen on creating headers to happen. I'll keep extra careful watch here and will try to make a GIF. Maybe it only occurs in more complex documents?

**Comment by jywarren** [comment]
This may need to be split into the bug search vs "converting back to plain text" as separate issues!

**Comment by keshav234156** [comment]
@jywarren Can you please share the problem related to Header that needs to be fixed here?

**Comment by shreyaa-sharmaa** [comment]
The most practical way to implement it in my opinion is to switch back to plain text after the max heading is reached. 
I agree with @jywarren this need to be included in the doc for users so that they're aware of all the features.

**Comment by shreyaa-sharmaa** [comment]
For Heading bug search this is all that I found so far
![PublicLab Editor-Google-Chrome-2020-06-13-23-20-42](https://user-images.githubusercontent.com/66686803/84576400-1bd67b80-add2-11ea-9f23-220d9141af01.gif)

The same actions yield the desired result in markdown mode.

**Comment by jywarren** [comment]
It does look like that's a good documentation of a bug. I wonder if we can now write a Jest test to reproduce what's happening in that gif!

**Comment by jywarren** [comment]
I also found this. I'm not sure, is it a related bug or shall we split this into two? Basically i try deleting all text and writing, but if the text started with a title, i can't seem to get out of title mode. 

This makes for a really great test though! It's so easy to reproduce in Jest. Start with some text that includes a header. Then press select all and delete. Then try typing again, and write an assertion that it should not be a title text!

![editor2](https://user-images.githubusercontent.com/24359/88101136-90d46800-cb6b-11ea-8e15-e78bd8eddc86.gif)

@keshav234156 would you like to try writing a test for this?
